### PR TITLE
dialects: (riscv_cf) add riscv_cf dialect

### PR DESCRIPTION
**Pull request by superlopuh**
This adds the riscv_cf dialect, duplicating some instructions in the riscv dialect, but with dataflow semantics with successors etc. This lets us reason about dataflow using xDSL IR constructs, and is a step towards being able to interpret exactly the same IR that we use to print assembly.

I forgot to add two things, will do them tomorrow:
 - [x] attributes
 - [x] comments in assembly

### CODE REVIEW

File: xdsl/dialects/riscv_cf.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    if (x[rs1] >=u x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#bgeu
+    """
+
+    name = "riscv_cf.bgeu"
+
+
+@irdl_op_definition
+class BranchOp(IRDLOperation, riscv.RISCVOp):
```
Comment (superlopuh) [inline xdsl/dialects/riscv_cf.py:247]: Taking suggestions for names for this operation, which is basically a no-op. Its use is to fall into a block that can be jumped to as well.

Comment (AntonLydike) [reply xdsl/dialects/riscv_cf.py:247]: Hmmm, how about `yield_through`?

**Review by AntonLydike — APPROVED** [review]
Nice work!

File: xdsl/dialects/riscv_cf.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/riscv_cf.py:258]: Oops, forgot to do this, let's rename later if we feel like it

**Pull request merged**